src/Features/CSharp/Portable/RemoveUnusedVariable/CSharpRemoveUnusedVariableCodeFixProvider.cs

```diff
@@ -69,5 +69,19 @@ protected override void RemoveOrReplaceNode(SyntaxEditor editor, SyntaxNode node
 
         protected override SeparatedSyntaxList<SyntaxNode> GetVariables(LocalDeclarationStatementSyntax localDeclarationStatement)
             => localDeclarationStatement.Declaration.Variables;
+
+        protected override bool ShouldOfferFix(ISyntaxFactsService syntaxFacts, SyntaxNode node)
+        {
+            // Local declarations must be parented by an executable block (or global statement) otherwise
+            // removing them would be invalid (and more than likely crash the fixer)
+            if (node.Parent.Parent is LocalDeclarationStatementSyntax { Parent: var localDeclParent } &&
```

Comment:
insead of PArent.Parent, i would find the ancestor that is a TLocalDeclarationStatement.  That's what the fixer actually does.  in general, we want the logic to be the same so we know that whatever the fix logic is deciding to remove the checking logic has checked the same thing.

Reply:
the relevant line is: https://github.com/dotnet/roslyn/pull/56902/files#diff-1a572f8959b2a2b068b03239a019de5818b4a1a57868cb49c62d84baf37ac17fL114

Consider extracting this out to a common helper to keep them in sync.  Once you find the LocalDecl, then you check that it's either in a GlobalStatement or an ExecutableBlock.